utils/downloads.py

```diff
@@ -53,3 +54,30 @@ def download_sounds(licenses_url, pack):
     response = HttpResponse(filelist, content_type="text/plain")
     response['X-Archive-Files'] = 'zip'
     return response
+
+
+def should_suggest_donation(user, times_displayed):
+    """
+    This method indicates when we should display the donation popup to the
+    user. This will be based on 3 settings indicating how many days
+    after a donation we show the popup again, after how many downloads we
+    display the popup and for how long. The popup will be shown a number of
+    times per day.
+    """
+
+    if times_displayed and times_displayed > settings.DONATION_MODAL_DISPLAY_TIMES_DAY:
+        return False
+
+    last_donation = user.donation_set.order_by('created').last()
+    period = datetime.datetime.now()\
+            - datetime.timedelta(days=settings.DONATION_MODAL_DOWNLOAD_DAYS)
+    num_downloads_in_period = Download.objects.filter(user=user,
```

Comment:
I'd say there was a problem with the naming. What the cookie was supposed to count was the number of times the modal was shown, not really the number of downloads. I, https://github.com/MTG/freesound/pull/820/commits/cef84553675c026d7b28c25ca8413209246ece85 I refactored this code and renamed stuff so now I think its less confusing.